packages/modules/cra-template-web-viewer/template/src/App.tsx

```diff
@@ -3,31 +3,31 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
 import "./App.scss";
 
-import { BrowserAuthorizationClientConfiguration } from "@bentley/frontend-authorization-client";
 import {
-  FitViewTool,
-  IModelApp,
-  ScreenViewport,
-  StandardViewId,
-} from "@bentley/imodeljs-frontend";
-import { Viewer } from "@itwin/web-viewer-react";
-import React, { useEffect, useState } from "react";
-
-import { Header } from "./Header";
+  BrowserAuthorizationCallbackHandler,
+  BrowserAuthorizationClient,
+  FrontendAuthorizationClient,
+} from "@bentley/frontend-authorization-client";
+import { FrontendRequestContext } from "@bentley/imodeljs-frontend";
+import React, { useContext } from "react";
+
+import AuthorizationOptions from "./AuthorizationOptions";
+import IModelJsContext from "./Context";
 import { history } from "./history";
+import ViewerStartup from "./ViewerStartup";
 
 const App: React.FC = () => {
-  const [isAuthorized, setIsAuthorized] = useState(
-    (IModelApp.authorizationClient?.hasSignedIn &&
-      IModelApp.authorizationClient?.isAuthorized) ||
-      false
-  );
-  const [isLoggingIn, setIsLoggingIn] = useState(false);
-  const [iModelId, setIModelId] = useState(process.env.IMJS_IMODEL_ID);
-  const [contextId, setContextId] = useState(process.env.IMJS_CONTEXT_ID);
+  // This will be the default context read from Context.tsx
+  const context = useContext(IModelJsContext);
```

Comment:
you're calling useContext without wrapping the parent level component with the provider

Reply:
I was essentially using it as a global variable, which i thought was fine to limit the amount of code. Provider is added so it doesn't look like magic now.